src/devices/Servo/ServoMotor.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Device.Gpio;
+using System.Device.Pwm;
+using System.Device.Pwm.Drivers;
+
+namespace Iot.Device.Servo
+{
+    public class ServoMotor : IDisposable
+    {
+        private int servoPin;
+        private int pwmChannel;
+        private double PulseFrequency = 20.0;
+
+        double currentPulseWidth = 0;
+
+        private float angle;
+        /// <summary>
+        /// Servo motor definition.
+        /// </summary>
+        private ServoMotorDefinition definition;
+        /// <summary>
+        /// The duration per angle's degree.
+        /// </summary>
+        private double rangePerDegree;
+        private PwmController pwmController;
+
+        public ServoMotor(int pinNumber, int pwmChannel, ServoMotorDefinition definition)
+        {                       
+            this.definition = definition;
+            PulseFrequency = definition.Period / 1000.0;
+
+            UpdateRange();
+
+            servoPin = pinNumber;
+            this.pwmChannel = pwmChannel;
+            // try hardware PWM first
+            try{
+                pwmController = new PwmController();
+                pwmController.OpenChannel(pinNumber, pwmChannel);
+                Console.WriteLine("Using hardware PWM");
+
+            } catch
+            {
+                pwmController = new PwmController(new SoftPwm(true));
+                pwmController.OpenChannel(pinNumber, pwmChannel);
+                Console.WriteLine("Using software PWM");
+            }
+
+            pwmController.StartWriting(servoPin, pwmChannel, 1000 / PulseFrequency,  (1 - (PulseFrequency - currentPulseWidth)/PulseFrequency) * 100);         
+
+        }
+
+        /// <summary>
+        /// Rotate the servomotor with a specific pulse in microseconds
+        /// </summary>
+        /// <param name="duration">Pulse duration in microseconds</param>
+        public void SetPulse(uint duration)
```

Comment:
I'd put unit also in the variable name (i.e. durationUs or durationMicroseconds)